oshdb-api/src/main/java/org/heigit/ohsome/oshdb/api/mapreducer/GeometrySplitter.java

```diff
@@ -197,19 +201,32 @@ D getData() {
             boolean intersectsAfter = contributionGeometryAfter != null
                 && pg.intersects(contributionGeometryAfter);
             if ((!intersectsBefore) && (!intersectsAfter)) {
-              return Stream.empty(); // not actually intersecting -> skip
+              // not actually intersecting -> skip
+              return Stream.empty();
             } else {
               return Stream.of(new IndexData<>(index, new OSMContribution(data,
-                  new LazyEvaluatedObject<>(() -> poop.intersection(contributionGeometryBefore)),
-                  new LazyEvaluatedObject<>(() -> poop.intersection(contributionGeometryAfter)))
+                  new LazyEvaluatedObject<>(() ->
+                      faultTolerantIntersection(contributionGeometryBefore, poop)),
+                  new LazyEvaluatedObject<>(() ->
+                      faultTolerantIntersection(contributionGeometryAfter, poop)))
               ));
             }
           } catch (TopologyException ignored) {
-            return Stream.empty(); // JTS cannot handle broken osm geometry -> skip
+            // JTS cannot handle broken osm geometry -> skip
+            return Stream.empty();
           }
         }).collect(Collectors.toMap(IndexData::getIndex, IndexData::getData));
   }
 
+  private static Geometry faultTolerantIntersection(Geometry subject, FastPolygonOperations poop) {
+    try {
+      return poop.intersection(subject);
+    } catch (TopologyException ignored) {
+      // JTS cannot handle broken osm geometry -> return empty geometry
+      return (new GeometryFactory()).createGeometryCollection();
```

Comment:
I assume it's irrelevant which `Geometry` you use here? Or is there a specific reason you use `GeometryCollection`?

Reply:
I also thought about using the geometry type of `subject` to return a matching empty geometry: mostly/only(?) empty polygon or empty geometry collection, but am not sure if it's worth the additional complexity. My reasoning for choosing the most GeometryCollection type is that broken polygons could also end up as geometry collections if our geometry building algorithm was a bit smarter to detect invalid geometries and used the same [fall back](https://github.com/GIScience/oshdb/blob/master/oshdb-util/src/main/java/org/heigit/ohsome/oshdb/util/geometry/OSHDBGeometryBuilder.java#L136) as it does for other error cases.

Reply:
thanks for the explanation :+1: